### PR TITLE
[ZEPPELIN-870] Improvement build and pom normalized

### PR DESCRIPTION
**Pull request by shijinkui**
### What is this PR for?
1. frontend-maven-plugin upgrade to 1.0, as low version occur error when build on osx
2. upgrade spark version from 1.4.1 to 1.6.1
3. define maven dependency scala version as property, used by  dependency version include scala version 
4. dependencyManagement define junit and scala-test dependency, same to pluginManagement
5. add `spark.iscached` property, used by  profile when need download spark dist by download-maven-plugin. 
6. add spark-2.0 and hadoop-2.7 profile

### What type of PR is it?
[Improvement]

### What is the Jira issue?
 [ZEPPELIN-870](https://issues.apache.org/jira/browse/ZEPPELIN-870) 

### How should this be tested?

run build command:
`export MAVEN_OPTS="-Xmx3g -XX:MaxPermSize=512M -XX:ReservedCodeCacheSize=512M"; mvn clean package -Pbuild-distr -DskipTests -Pspark-1.6 -Phadoop-2.7 -Ppyspark -Pvendor-repo -Dignite.version=1.1.0-incubating -Pscalding -Pspark-dist-cached -Drat.skip=true`

### CODE REVIEW

**Comment by shijinkui** [comment]
who can review this pr? @catap @jonbuffington

File: spark-dependencies/pom.xml [hunk 1]
```diff
@@ -36,7 +37,7 @@
 
 
   <properties>
-    <spark.version>1.4.1</spark.version>
+    <spark.version>1.6.1</spark.version>
```
Comment (Leemoonsoo) [inline spark-dependencies/pom.xml]: Thanks @shijinkui for the contribution. I want to point out that change spark.version here is probably not the good way to make 1.6.1 as a default build. while all other dependency versions are remained for spark 1.4.1.

I think activating profile `spark-1.6` by default, could be a possible way to make 1.6.1 default.

Comment (shijinkui) [inline spark-dependencies/pom.xml]: good idea

File: spark-dependencies/pom.xml [hunk 2]
```diff
@@ -776,6 +818,14 @@
     </profile>
 
     <profile>
+      <!--only used for pyspark/sparkR needed spark dist is downloaded, no need to download auto-->
+      <id>spark-dist-cached</id>
```
Comment (bzz) [inline spark-dependencies/pom.xml]: Introducing such profile totally makes sense to enable opt-out behavior when such things as spark are not needed.

But I think we need to make sure it is `activeByDefault ` and point to `validate` phase, to have a build behavior compatible with current one, as our CI infrastructure relies on it for tests.

File: zeppelin-web/pom.xml [hunk 3]
```diff
@@ -33,9 +33,9 @@
   <name>Zeppelin: web Application</name>
 
   <!-- See https://github.com/eirslett/frontend-maven-plugin/issues/229 -->
-  <prerequisites>
-    <maven>3.1.0</maven>
-  </prerequisites>
+  <!--<prerequisites>-->
+    <!--<maven>3.1.0</maven>-->
```
Comment (bzz) [inline zeppelin-web/pom.xml]: Is there a reason to remove this one? 

Quick search\blame for this line points to [ZEPPELIN-622](https://github.com/apache/incubator-zeppelin/commit/48f875d42b90b6b175aa5deed2a552af8183e4a4) as a reason it was introduced in the first place

**Comment by bzz** [comment]
Thank you for contribution!

It is very sound and useful, please see the comments above for more detailed feedback.

The only thing that can be improved to merge it faster is - if we could separate it on 2 parts: 
 1. "no side-effects" one: normalization&variable reconciliation, which does not change or affect any build behavior
 2. other changes affecting\optimizing the build, that do have side effects (add phases\change defaults\etc)

that would simplify the review and make things even smoother.

File: spark-dependencies/pom.xml (hunk 2 above)
Comment (shijinkui) [inline spark-dependencies/pom.xml]: validate phrase is OK. only make sure let download-maven-plugin don't trigger download as it's so slow developer can download spark dist manual.

File: zeppelin-web/pom.xml (hunk 3 above)
Comment (shijinkui) [inline zeppelin-web/pom.xml]: no need use low maven version for avoid Ubuntu 15.04's problem itself'. https://github.com/eirslett/frontend-maven-plugin/issues/229  , upgrade the frontend-maven-plugin to 1.0

Comment (shijinkui) [inline zeppelin-web/pom.xml]: use maven 3.3.9 is ok on mac. ubuntu 15.04 not test ye.

**Comment by shijinkui** [comment]
hi, @bzz  thanks for review. had add spark-1.6 default active and validate phrase. 
For the changes not more enough to separate, I think in one pr may be suitable.

>I looked though most of code, the code format of java and  scala code not well-formed. I'll have a new issue to refactor, that will be more code changes. :)

**Comment by shijinkui** [comment]
hi, @bzz the bower error, my local compile is ok, how can i continue

```
[ERROR] bower angular-ui-ace#0.1.1        invalid-meta angular-ui-ace is missing "ignore" entry in bower.json
[ERROR] bower ng-focus-if#~1.0.2              mismatch Version declared in the json (1.0.4) is different than the resolved one (1.0.5)
[ERROR] bower angular-websocket#~1.0.13       mismatch Version declared in the json (1.0.13) is different than the resolved one (1.0.14)
[ERROR] bower ace-builds#1.1.9            invalid-meta ace-builds is missing "main" entry in bower.json
[ERROR] bower ace-builds#1.1.9            invalid-meta ace-builds is missing "ignore" entry in bower.json
[ERROR] bower perfect-scrollbar       extra-resolution Unnecessary resolution: perfect-scrollbar#~0.5.4
```

File: spark-dependencies/pom.xml (hunk 2 above)
Comment (bzz) [inline spark-dependencies/pom.xml]: If we are changing the default behaviour (download&cache spark) we need to: 
 - update CI scripts 
 - update release docs 
as they rely on current behavior of spark been downloaded.

I would suggest we keep the default behavior: it downloads and caches spark in local FS and it's slow only once (download plugin have [ovewrite flag](https://github.com/maven-download-plugin/maven-download-plugin/blob/master/download-maven-plugin/src/main/java/com/googlecode/download/maven/plugin/internal/WGet.java#L59) `false` by default) but allow developers, who do not need it at all, to explicitly opt-out this though the profie.

File: zeppelin-web/pom.xml (hunk 3 above)
Comment (bzz) [inline zeppelin-web/pom.xml]: I'm not sure I understand what you mean, but I do not see a reason to remote enforcement of the minimum maven version. It was added there to fail build fast and avoid mentioned bug.

**Comment by bzz** [comment]
@shijinkui I do not think that mentioned fronted build warnings are relevant to CI build failure.
It seems that Selenium integration tests are failing

```
Caused by: org.openqa.selenium.NoSuchElementException: Unable to locate element: {"method":"partial link text","selector":"Create new note"}
```

**Comment by bzz** [comment]
As #862 is merged - it should fix the CI, so could you please rebase on the latest master and see if that helps?

**Comment by shijinkui** [comment]
@bzz have rebased.  Failed all for the spark archive download timeout by the url `http://archive.apache.org/dist/spark/spark-1.6.1/spark-1.6.1-bin-without-hadoop.tgz`

```
https://travis-ci.org/apache/incubator-zeppelin/jobs/133047225
[INFO] --- download-maven-plugin:1.2.1:wget (download-sparkr-files) @ zeppelin-spark-dependencies ---
Downloading: http://archive.apache.org/dist/spark/spark-1.6.1/spark-1.6.1-bin-without-hadoop.tgz
org.apache.maven.wagon.providers.http.httpclient.conn.HttpHostConnectException: Connect to archive.apache.org:80 [archive.apache.org/163.172.17.199] failed: Connection refused


Downloading: http://archive.apache.org/dist/spark/spark-1.5.2/spark-1.5.2-bin-without-hadoop.tgz
org.apache.maven.wagon.providers.http.httpclient.conn.HttpHostConnectException: Connect to archive.apache.org:80 [archive.apache.org/163.172.17.199] failed: Connection refused
	at org.apache.maven.wagon.providers.http.httpclient.impl.conn.HttpClientConnectionOperator.connect(HttpClientConnectionOperator.java:140)
	at 

https://travis-ci.org/apache/incubator-zeppelin/jobs/133047227
```

**Comment by jongyoul** [comment]
@shijinkui It becomes green. Could you please trigger it again?

**Comment by shijinkui** [comment]
@jongyoul I have no authorization。 cc @bzz

**Comment by jongyoul** [comment]
@shijinkui Just close and reopen it. :-)

**Comment by jongyoul** [comment]
I agree with @bzz. If your PR has both of parts which change behavior or not, I suggest you divide two seperate PR because of easy review. If you mix them, all reviewers should judge everyline which is normallized part or changing something. I also suggest you contribute changing style at first, and changing behavior is the second one. How about you?

**Pull request closed without merging**